Custom-Pylint-checks.md

```diff
@@ -0,0 +1,598 @@
+        ),
+    }
+```
+
+Here's what we've defined so far:
+
+* `__implements__`: Specifies what kind of checker we are creating. In this case, we are creating an AST checker.
+* `name`: Short name of the checker that will be used in the configuration file.
+* `priority`: A negative integer indicating when the check should run. Checkers run from the most negative priority to most positive.
+* `msgs`: A dictionary specifying what messages this checker can raise. The dictionary has the following form:
+
+  ```python
+  {
+      'message id': ('displayed message', 'message symbol', 'message help')
+  }
+  ```
+
+  * **Message ID**: A unique identifier for the message consisting of a letter followed by 4 digits. The letter is the first letter of the message category, and the available categories are Convention, Warning, Error, Fatal, and Refactoring. The first two digits must be the same for all of the checker's messages.
+
+    Be sure to check that your message ID isn't already used by one of [Pylint's existing rules](https://pylint.pycqa.org/en/latest/technical_reference/features.html) or one of [our custom rules](https://github.com/oppia/oppia/blob/develop/scripts/linters/pylint_extensions.py).
+
+  * **Displayed message**: The error message displayed to the user.
+
+  * **Message symbol**: An alias of the message ID that can be used wherever the ID can be used. For example, to disable a message with a code comment, you can specify either the ID or the symbol.
+
+  * **Message help**: Displayed when the user runs `pylint --help-msg` to learn more about your message.
+
+You can also set an `options` variable to specify options that can be set by the configuration file and read by your rule. See the [Pylint documentation](https://pylint.pycqa.org/en/latest/how_tos/custom_checkers.html#writing-an-ast-checker) for details.
+
+Next, you need to write the checker code. Pylint lets you define methods that get called as it traverses the AST. Methods of the form `visit_nodename()` are called upon entering the subtree rooted at a node of type `nodename`, and methods of the form `leave_nodename()` are called upon leaving the subtree. Each method is called with a single argument--the node of type `nodename`.
+
+For this case, you can define a method `visit_const()` that will be called whenever Pylint reaches a `Const` node. Then you can check the node's value to see if it equals `"Hello, world!"`. The method looks like this:
+
+```python
+def visit_const(self, node):
+    if node.value == "Hello, world!":
+        self.add_message(
+            'hello-world', node=node)
+```
+
+That's it! After working through the two other kinds of linters, we'll discuss how to run and test your checker.
+
+If you want an alternative tutorial for writing an AST checker, take a look at [the one Pylint provides](https://pylint.pycqa.org/en/latest/how_tos/custom_checkers.html#writing-an-ast-checker).
+
+#### Write the token checker
```

Comment:
I'm actually a bit lost in terms of the structure of this doc. Are you trying to write the same lint check three different ways?

I don't think there's enough orientation being given to the reader here; you might want to tell them in advance that this is what we're doing.

Reply:
I added this sentence at the beginning:

> Below, we'll walk through how to implement this rule using each of the three kinds of checkers in Pylint: AST checkers, token checkers, and raw checkers.